src/vision.cs

```diff
@@ -3872,4 +4220,267 @@ interface VNDetectDocumentSegmentationRequest {
+		CVPixelBuffer InstanceMask { get; }
+
+		[Export ("allInstances", ArgumentSemantic.Copy)]
+		NSIndexSet AllInstances { get; }
+
+		[Export ("generateMaskForInstances:error:")]
+		[return: NullAllowed]
+		CVPixelBuffer GenerateMask (NSIndexSet instances, [NullAllowed] out NSError error);
+
+		[Export ("generateMaskedImageOfInstances:fromRequestHandler:croppedToInstancesExtent:error:")]
+		[return: NullAllowed]
+		CVPixelBuffer GenerateMaskedImage (NSIndexSet instances, VNImageRequestHandler requestHandler, bool cropResult, [NullAllowed] out NSError error);
+
+		[Export ("generateScaledMaskForImageForInstances:fromRequestHandler:error:")]
+		[return: NullAllowed]
+		CVPixelBuffer GenerateScaledMask (NSIndexSet instances, VNImageRequestHandler requestHandler, [NullAllowed] out NSError error);
+	}
+
+	[TV (17, 0), Mac (14, 0), iOS (17, 0), MacCatalyst (17, 0)]
+	[BaseType (typeof (VNRecognizedPoints3DObservation))]
+	[DisableDefaultCtor]
+	interface VNHumanBodyPose3DObservation {
+		[Export ("heightEstimation")]
+		VNHumanBodyPose3DObservationHeightEstimation HeightEstimation { get; }
+
+		[Export ("cameraOriginMatrix")]
+		Vector4 CameraOriginMatrix {
+			[MarshalDirective (NativePrefix = "xamarin_simd__", Library = "__Internal")]
+			get;
+		}
+
+		[Export ("availableJointsGroupNames", ArgumentSemantic.Copy)]
+		[BindAs (typeof (VNHumanBodyPose3DObservationJointsGroupName []))]
+		NSString [] AvailableJointsGroupNames { get; }
+
+		[Export ("availableJointNames", ArgumentSemantic.Copy)]
+		[BindAs (typeof (VNHumanBodyPose3DObservationJointName []))]
+		NSString [] AvailableJointNames { get; }
+
+		[Export ("bodyHeight")]
+		float BodyHeight { get; }
+
+		[Export ("recognizedPointsForJointsGroupName:error:")]
+		[return: NullAllowed]
+		NSDictionary<NSString, VNHumanBodyRecognizedPoint3D> GetRecognizedPoints ([BindAs (typeof (VNHumanBodyPose3DObservationJointsGroupName))] NSString jointName, [NullAllowed] out NSError error);
+
+		[Export ("recognizedPointForJointName:error:")]
+		[return: NullAllowed]
+		VNHumanBodyRecognizedPoint3D GetRecognizedPoint ([BindAs (typeof (VNHumanBodyPose3DObservationJointName))] NSString jointName, [NullAllowed] out NSError error);
+
+		[Export ("pointInImageForJointName:error:")]
+		[return: NullAllowed]
+		VNPoint GetPointInImage ([BindAs (typeof (VNHumanBodyPose3DObservationJointName))] NSString jointName, [NullAllowed] out NSError error);
+
+		[Export ("parentJointNameForJointName:")]
+		[return: NullAllowed]
+		[return: BindAs (typeof (VNHumanBodyPose3DObservationJointName))]
+		NSString GetParentJointName ([BindAs (typeof (VNHumanBodyPose3DObservationJointName))] NSString jointName);
+
+		[Export ("getCameraRelativePosition:forJointName:error:")]
+		[MarshalDirective (NativePrefix = "xamarin_simd__", Library = "__Internal")]
+		bool GetCameraRelativePosition (out Vector4 modelPositionOut, [BindAs (typeof (VNHumanBodyPose3DObservationJointName))] NSString jointName, [NullAllowed] out NSError error);
```

Comment:
The build and tests are passing (no weird xtro errors!), but would appreciate any insight on how to improve the Marshal Directive support..

In the binding, the GetCameraRelativePosition selector introduced a new native function referenced by its MarshalDirective attribute, so the signature entry was added to bindings-generator.cs. But the tricky part is the ref vs out parameter distinction. 

From the binding highlighted, this is what gets generated:
`_xamarin_simd__bool_objc_msgSend_out_Vector4_NativeHandle_ref_NativeHandle`

In this case both the Vector4 modelPositionOut and NSError error params should be out since there is no need for them to be init prior to using the method ([ref](https://developer.apple.com/documentation/vision/vnhumanbodypose3dobservation/4210411-getcamerarelativeposition)).

But the way the generator works is that when a NSError param, is declared as an out, the signature is generated as a ref. 

For ex, this signature in src/gamecontroller.cs:
 
```
[Export ("acquireDeviceWithError:")]
bool AcquireDevice ([NullAllowed] out NSError error);
```

gets generated as:
`ret = global::ObjCRuntime.Messaging.bool_objc_msgSend_ref_IntPtr (this.Handle, selAcquireDeviceWithError_XHandle, &errorValue);`

Notice the ref_IntPtr

Based on my understanding, this isn’t really a problem because the out vs. ref discrepancy only exists in the .NET world, and when invoking into Obj C the difference is not there?

But what makes this complicated is that the out param is taken into consideration for the Vector4 modelPositionOut parameter however, which is why the isOut param logic was added else we could just disregard the out vs ref discrepancy and treat both cases identically? Also afaik there is not another case in the repo of the Marshal Directive being use in conjunction with an out param, so maybe makes sense for this support to be needed now?